deluge/core/torrentmanager.py

```diff
@@ -384,25 +382,31 @@ def add(self, torrent_info=None, state=None, options=None, save_state=True,
         if options["seed_mode"]:
             add_torrent_params["flags"] |= lt.add_torrent_params_flags_t.flag_seed_mode
 
-        # We need to pause the AlertManager momentarily to prevent alerts
-        # for this torrent being generated before a Torrent object is created.
-        component.pause("AlertManager")
-
+        d = Deferred()
         try:
-            handle = self.session.add_torrent(add_torrent_params)
-            if not handle.is_valid():
-                raise InvalidTorrentError("Torrent handle is invalid!")
+            self.torrents_loading[torrent_id] = (d, options, state, filename, magnet, resume_data, filedump, save_state)
```

Comment:
Should we check if torrent_id is already in torrents_loading to prevent a possible race condition adding the same torrent multiple times very quickly?